components/jupyter-web-app/backend/kubeflow_jupyter/common/base_app.py

```diff
@@ -25,8 +27,20 @@ def get_notebooks(namespace):
     if not data["success"]:
         return jsonify(data)
 
-    data["notebooks"] = [utils.process_resource(nb)
-                         for nb in data["notebooks"]["items"]]
+    items = []
+    for nb in data["notebooks"]["items"]:
+        nb_name = nb["metadata"]["name"]
+        nb_creation_time = dt.datetime.strptime(
+            nb["metadata"]["creationTimestamp"], "%Y-%m-%dT%H:%M:%SZ")
+        nb_events = api.get_notebook_events(namespace, nb_name)
+        nb_events = nb_events["notebook-events"].items if nb_events["success"] else []
+        # User can delete and then create a nb server with the same name
+        # Make sure previous events are not taken into account
+        nb_events = filter(lambda e: utils.event_timestamp(e) >= nb_creation_time,
+                           nb_events)
```

Comment:
Doesn't Kubernetes delete the Notebook's Events once the Notebook instance gets deleted?

Reply:
No. You can try to create a notebook and then delete it and you will see that the pods and statefulsets events are not deleted either. However, if a new controller starts and tries to watch events from the pod or the ss, it won't see events since they are associated with a resource that no longer exists. But if you ask the master-API they are still there.

Reply:
I see, it makes sense then to filter the events and show only the ones that should be for each CR's current life-cycle and not from previous ones (based on the timestamps of the CR and the Events). ACK